File: MD_sample/Python_imperative/verlet.py
from md    import Npt, box, Particle
from force import dnint

def vverlet_1(q,dt):
    for i in range (Npt):
        for k in range (3):
            q[i].xx[k] = q[i].xx[k] + 0.5 * dt * dt * q[i].ff[k]/q[i].xm
            q[i].xx[k] = q[i].xx[k] - box * dnint(q[i].xx[k]/box)
            q[i].vv[k] = q[i].vv[k] + 0.5 * dt * q[i].ff[k]/q[i].xm
            
def vverlet_2(q,dt):
    for i in range (Npt):
        for k in range (3):
            q[i].vv[k] = q[i].vv[k] + 0.5 * dt * q[i].ff[k]/q[i].xm
            

File: MD_sample/C_procedural/verlet.c
#include "md.h"

double dnint(double x);

/* First step of velocity verlet */
void vverlet_1(particle *q, double dt)
{
  unsigned i, k;
  for (i=0; i<Npt; i++) {
    for (k=0; k<3; k++) {
      q[i].xx[k] += dt * q[i].vv[k] + 0.5*dt*dt*q[i].ff[k]/q[i].xm;
      q[i].xx[k] -= box * dnint(q[i].xx[k]/box);
      q[i].vv[i] += 0.5*dt*q[i].ff[k]/q[i].xm;
    }
  }
}

/* Second step of velocity verlet */
void vverlet_2(particle *q, double dt)
{
  unsigned i, k;
  for (i=0; i<Npt; i++) {
    for (k=0; k<3; k++) {
      q[i].vv[i] += 0.5*dt*q[i].ff[k]/q[i].xm;
    }
  }
}

